frontend/src/components/accountPage/SSHKeysForm/SSHKeysForm.tsx

```diff
@@ -0,0 +1,125 @@
+import { Input, Form, Button, Row } from 'antd';
+import { FC, useState } from 'react';
+
+export interface ISSHKeysFormProps {
+  onSaveKey: (newKey: { name: string; key: string }) => void;
+  onCancel: () => void;
+}
+
+const acceptedAlgorithms = [
+  'sk-ecdsa-sha2-nistp256@openssh.com',
+  'ecdsa-sha2-nistp256',
+  'ecdsa-sha2-nistp384',
+  'ecdsa-sha2-nistp521',
+  'sk-ssh-ed25519@openssh.com',
+  'ssh-ed25519',
+  'ssh-dss',
+  'ssh-rsa',
+];
+
+const SSHKeysForm: FC<ISSHKeysFormProps> = props => {
+  const { onSaveKey, onCancel } = props;
+  const [validForm, setValidForm] = useState<boolean>(false);
+  const [form] = Form.useForm();
+
+  const cancelForm = () => {
+    form.resetFields();
+    onCancel();
+  };
+  const validateSSHKey = async (rules: any, key: string) => {
+    if (!key) throw new Error('Key field is mandatory');
+    const result = key.split(/\s+/); // split regardless the amount of spaces
+    if (result.length !== 2 && result.length !== 3)
+      throw new Error('Invalid key format');
+    if (result[1] === '') throw new Error('Empty SSH key');
+    if (!acceptedAlgorithms.includes(result[0]))
+      throw new Error('Unrecognized SSH key format');
+  };
+
+  // Used to disable Save button
+  const handleChange = async () => {
```

Comment:
```suggestion
  const handleChange = async ({ name, key }: { name: string; key: string }) => {
```
If I'm not wrong, onValuesChange should receive parameters similarly to the submit. this should make the code cleaner and avoid the need to use getFieldValue

Reply:
I tried passing the `name` and `key` parameters to `handleChange` but I noticed some strange behavior. When `onValuesChange` (and handleChange) is triggered only the field that actually changes is set, the other is not. For example, if you write the key only the `key` parameter contains the typed value, `name` remains `undefined`.

Reply:
By looking at the code we have 
```ts
export interface Callbacks<Values = any> {
  //...
  onValuesChange?: (changedValues: any, values: Values) => void
}
```
so this means that you'd be able to do 
```ts
  const handleChange = async (_, { name, key }: { name: string; key: string }) => {
    //...
  }
```
(notice the first parameter that's just an underscore)
Have a try like this if you could

Reply:
Yes! Not it works thanks a lot :)